Blank white page in Firefox (clearing cookes+cache fixes)

### What went wrong?

https://openneuro.org/ just shows a blank white page in Firefox, but loads normally in Chromium and Vivaldi. Dataset views are similarly affected.

Browser console shows 4 requests to sentry.io being blocked on CORS grounds.

### Firefox console log:
<details>

Google Analytics and Tag Manager is being shimmed by Firefox. See https://bugzilla.mozilla.org/show_bug.cgi?id=1713687 for details. sandbox eval code:1:9
Cookie warnings 2
The value of the attribute “expires” for the cookie “_ga_MJNMY6D3FX” has been overwritten. [openneuro.org](https://openneuro.org/)
The value of the attribute “expires” for the cookie “_ga_MJNMY6D3FX” has been overwritten. 2 [openneuro.org](https://openneuro.org/)
XHRPOST
[https://o4507748938350592.ingest.us.sentry.io/api/4507748943724544/envelope/?sentry_version=7&sentry_key=988df6d6043a386440f614f1e936f73e&sentry_client=sentry.javascript.react/10.37.0](https://o4507748938350592.ingest.us.sentry.io/api/4507748943724544/envelope/?sentry_version=7&sentry_key=988df6d6043a386440f614f1e936f73e&sentry_client=sentry.javascript.react%2F10.37.0)
Tracking

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://o4507748938350592.ingest.us.sentry.io/api/4507748943724544/envelope/?sentry_version=7&sentry_key=988df6d6043a386440f614f1e936f73e&sentry_client=sentry.javascript.react%2F10.37.0. (Reason: CORS request did not succeed). Status code: (null).

The resource at “<URL>” was blocked because Enhanced Tracking Protection is enabled. 4
The resource at “https://o4507748938350592.ingest.us.sentry.io/api/4507748943724544/envelope/?sentry_version=7&sentry_key=988df6d6043a386440f614f1e936f73e&sentry_client=sentry.javascript.react%2F10.37.0” was blocked because Enhanced Tracking Protection is enabled.
[openneuro.org](https://openneuro.org/)
The resource at “https://o4507748938350592.ingest.us.sentry.io/api/4507748943724544/envelope/?sentry_version=7&sentry_key=988df6d6043a386440f614f1e936f73e&sentry_client=sentry.javascript.react%2F10.37.0” was blocked because Enhanced Tracking Protection is enabled.
[openneuro.org](https://openneuro.org/)
The resource at “https://o4507748938350592.ingest.us.sentry.io/api/4507748943724544/envelope/?sentry_version=7&sentry_key=988df6d6043a386440f614f1e936f73e&sentry_client=sentry.javascript.react%2F10.37.0” was blocked because Enhanced Tracking Protection is enabled.
[openneuro.org](https://openneuro.org/)
The resource at “https://o4507748938350592.ingest.us.sentry.io/api/4507748943724544/envelope/?sentry_version=7&sentry_key=988df6d6043a386440f614f1e936f73e&sentry_client=sentry.javascript.react%2F10.37.0” was blocked because Enhanced Tracking Protection is enabled.
[openneuro.org](https://openneuro.org/)
XHRPOST
[https://o4507748938350592.ingest.us.sentry.io/api/4507748943724544/envelope/?sentry_version=7&sentry_key=988df6d6043a386440f614f1e936f73e&sentry_client=sentry.javascript.react/10.37.0](https://o4507748938350592.ingest.us.sentry.io/api/4507748943724544/envelope/?sentry_version=7&sentry_key=988df6d6043a386440f614f1e936f73e&sentry_client=sentry.javascript.react%2F10.37.0)
Tracking

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://o4507748938350592.ingest.us.sentry.io/api/4507748943724544/envelope/?sentry_version=7&sentry_key=988df6d6043a386440f614f1e936f73e&sentry_client=sentry.javascript.react%2F10.37.0. (Reason: CORS request did not succeed). Status code: (null).

XHRPOST
[https://o4507748938350592.ingest.us.sentry.io/api/4507748943724544/envelope/?sentry_version=7&sentry_key=988df6d6043a386440f614f1e936f73e&sentry_client=sentry.javascript.react/10.37.0](https://o4507748938350592.ingest.us.sentry.io/api/4507748943724544/envelope/?sentry_version=7&sentry_key=988df6d6043a386440f614f1e936f73e&sentry_client=sentry.javascript.react%2F10.37.0)
Tracking

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://o4507748938350592.ingest.us.sentry.io/api/4507748943724544/envelope/?sentry_version=7&sentry_key=988df6d6043a386440f614f1e936f73e&sentry_client=sentry.javascript.react%2F10.37.0. (Reason: CORS request did not succeed). Status code: (null).

XHRPOST
[https://o4507748938350592.ingest.us.sentry.io/api/4507748943724544/envelope/?sentry_version=7&sentry_key=988df6d6043a386440f614f1e936f73e&sentry_client=sentry.javascript.react/10.37.0](https://o4507748938350592.ingest.us.sentry.io/api/4507748943724544/envelope/?sentry_version=7&sentry_key=988df6d6043a386440f614f1e936f73e&sentry_client=sentry.javascript.react%2F10.37.0)
Tracking

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://o4507748938350592.ingest.us.sentry.io/api/4507748943724544/envelope/?sentry_version=7&sentry_key=988df6d6043a386440f614f1e936f73e&sentry_client=sentry.javascript.react%2F10.37.0. (Reason: CORS request did not succeed). Status code: (null).

​</details>

Those errors don't show up in Chromium or Vivaldi dev consoles (though different errors do):

### Vivaldi console log:
<details>

www.googletagmanager.com/gtag/js:1  Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
redactor-32c3e863ebfcba166eb4c2604f1c70b71422765999a6c2a275a3668dd99c4834.js:3

Error checking advanced CSAT feature flag: ReferenceError: DataStore is not defined
    at i._isExtendedAdvancedCsatEnabled (redactor-32c3e863ebfcba166eb4c2604f1c70b71422765999a6c2a275a3668dd99c4834.js:3:30984)
    at i.init (redactor-32c3e863ebfcba166eb4c2604f1c70b71422765999a6c2a275a3668dd99c4834.js:1:31195)
    at new i (redactor-32c3e863ebfcba166eb4c2604f1c70b71422765999a6c2a275a3668dd99c4834.js:1:27421)
    at HTMLTextAreaElement.<anonymous> (redactor-32c3e863ebfcba166eb4c2604f1c70b71422765999a6c2a275a3668dd99c4834.js:1:13169)
    at o.each (fdwidget-aa5e83d012ee0d82d8b2cc675b15c699808b950c6933f083d650003ab4a79d13.js:4:5737)
    at init.each (fdwidget-aa5e83d012ee0d82d8b2cc675b15c699808b950c6933f083d650003ab4a79d13.js:4:2327)
    at jQuery.fn.redactor (redactor-32c3e863ebfcba166eb4c2604f1c70b71422765999a6c2a275a3668dd99c4834.js:1:13095)
    at HTMLDocument.<anonymous> (fdwidget-aa5e83d012ee0d82d8b2cc675b15c699808b950c6933f083d650003ab4a79d13.js:18:4042)
    at d (fdwidget-aa5e83d012ee0d82d8b2cc675b15c699808b950c6933f083d650003ab4a79d13.js:4:8572)
    at Object.fireWith (fdwidget-aa5e83d012ee0d82d8b2cc675b15c699808b950c6933f083d650003ab4a79d13.js:4:9338)
_isExtendedAdvancedCsatEnabled @ redactor-32c3e863ebfcba166eb4c2604f1c70b71422765999a6c2a275a3668dd99c4834.js:3


</details>

### Chromium console log:

<details>

redactor-32c3e863ebfcba166eb4c2604f1c70b71422765999a6c2a275a3668dd99c4834.js:3 Error checking advanced CSAT feature flag: ReferenceError: DataStore is not defined
    at i._isExtendedAdvancedCsatEnabled (redactor-32c3e863ebfcba166eb4c2604f1c70b71422765999a6c2a275a3668dd99c4834.js:3:30984)
    at i.init (redactor-32c3e863ebfcba166eb4c2604f1c70b71422765999a6c2a275a3668dd99c4834.js:1:31195)
    at new i (redactor-32c3e863ebfcba166eb4c2604f1c70b71422765999a6c2a275a3668dd99c4834.js:1:27421)
    at HTMLTextAreaElement.<anonymous> (redactor-32c3e863ebfcba166eb4c2604f1c70b71422765999a6c2a275a3668dd99c4834.js:1:13169)
    at o.each (fdwidget-aa5e83d012ee0d82d8b2cc675b15c699808b950c6933f083d650003ab4a79d13.js:4:5737)
    at init.each (fdwidget-aa5e83d012ee0d82d8b2cc675b15c699808b950c6933f083d650003ab4a79d13.js:4:2327)
    at jQuery.fn.redactor (redactor-32c3e863ebfcba166eb4c2604f1c70b71422765999a6c2a275a3668dd99c4834.js:1:13095)
    at HTMLDocument.<anonymous> (fdwidget-aa5e83d012ee0d82d8b2cc675b15c699808b950c6933f083d650003ab4a79d13.js:18:4042)
    at d (fdwidget-aa5e83d012ee0d82d8b2cc675b15c699808b950c6933f083d650003ab4a79d13.js:4:8572)
    at Object.fireWith (fdwidget-aa5e83d012ee0d82d8b2cc675b15c699808b950c6933f083d650003ab4a79d13.js:4:9338)
_isExtendedAdvancedCsatEnabled @ redactor-32c3e863ebfcba166eb4c2604f1c70b71422765999a6c2a275a3668dd99c4834.js:3
init @ redactor-32c3e863ebfcba166eb4c2604f1c70b71422765999a6c2a275a3668dd99c4834.js:1
i @ redactor-32c3e863ebfcba166eb4c2604f1c70b71422765999a6c2a275a3668dd99c4834.js:1
(anonymous) @ redactor-32c3e863ebfcba166eb4c2604f1c70b71422765999a6c2a275a3668dd99c4834.js:1
each @ fdwidget-aa5e83d012ee0d82d8b2cc675b15c699808b950c6933f083d650003ab4a79d13.js:4
each @ fdwidget-aa5e83d012ee0d82d8b2cc675b15c699808b950c6933f083d650003ab4a79d13.js:4
jQuery.fn.redactor @ redactor-32c3e863ebfcba166eb4c2604f1c70b71422765999a6c2a275a3668dd99c4834.js:1
(anonymous) @ fdwidget-aa5e83d012ee0d82d8b2cc675b15c699808b950c6933f083d650003ab4a79d13.js:18
d @ fdwidget-aa5e83d012ee0d82d8b2cc675b15c699808b950c6933f083d650003ab4a79d13.js:4
fireWith @ fdwidget-aa5e83d012ee0d82d8b2cc675b15c699808b950c6933f083d650003ab4a79d13.js:4
ready @ fdwidget-aa5e83d012ee0d82d8b2cc675b15c699808b950c6933f083d650003ab4a79d13.js:4
r @ fdwidget-aa5e83d012ee0d82d8b2cc675b15c699808b950c6933f083d650003ab4a79d13.js:4


</details>

### Expected behavior

the website works in Firefox

### How to reproduce

navigate to https://openneuro.org

### Desktop

```console
$ uname -sr
Linux 6.8.0-45-generic
$ firefox --version
Mozilla Firefox 140.7.1esr
$ chromium --version
Chromium 146.0.7680.80 snap
$ vivaldi --version
Vivaldi 7.8.3925.73 stable
```


### Phone

n/a

### Additional information

_No response_

--- a/packages/openneuro-app/src/scripts/authentication/google-redirect.tsx
+++ b/packages/openneuro-app/src/scripts/authentication/google-redirect.tsx
@@ -1,0 +1,26 @@
+import { useEffect } from "react"
+import { useLocation, useNavigate } from "react-router-dom"
+import { useUser } from "../queries/user"
+
+/**
+ * Redirects Google-authenticated users to the ORCID linking page.
+ * Renders nothing — just runs the redirect effect when the user query resolves.
+ */
+export function GoogleOrcidRedirect() {
+  const navigate = useNavigate()
+  const location = useLocation()
+  const { user, loading, error } = useUser()
+
+  useEffect(() => {
+    if (
+      !loading &&
+      !error &&
+      location.pathname !== "/orcid-link" &&
+      user?.provider === "google"
+    ) {
+      navigate("/orcid-link")
+    }
+  }, [location.pathname, user, loading, error, navigate])
+
+  return null
+}

--- a/packages/openneuro-app/src/scripts/authentication/profile.ts
+++ b/packages/openneuro-app/src/scripts/authentication/profile.ts
@@ -21,7 +21,11 @@ export const parseJwt = jwtDecode
 export function getProfile(cookies): OpenNeuroTokenProfile {
   const accessToken = cookies["accessToken"]
   if (!accessToken) return null
-  return parseJwt(accessToken) as OpenNeuroTokenProfile
+  try {
+    return parseJwt(accessToken) as OpenNeuroTokenProfile
+  } catch {
+    return null
+  }
 }
 /**
  * Return profile if token is not expired.

--- a/packages/openneuro-app/src/scripts/index.tsx
+++ b/packages/openneuro-app/src/scripts/index.tsx
@@ -1,5 +1,4 @@
-import React, { useEffect } from "react"
-import { useLocation, useNavigate } from "react-router-dom"
+import React from "react"
 import Uploader from "./uploader/uploader"
 import AppRoutes from "./routes"
 import HeaderContainer from "./common/containers/header"
@@ -7,37 +6,12 @@ import FooterContainer from "./common/containers/footer"
 import { SearchParamsProvider } from "./search/search-params-ctx"
 import { UserModalOpenProvider } from "./utils/user-login-modal-ctx"
 import { useAnalytics } from "./utils/analytics"
-import { useUser } from "./queries/user"
 import { NotificationsProvider } from "./users/notifications/user-notifications-context"
+import { GoogleOrcidRedirect } from "./authentication/google-redirect"
 import "../assets/email-header.png"
 
 const Index: React.FC = () => {
   useAnalytics()
-
-  const navigate = useNavigate()
-  const location = useLocation()
-  const { user, loading, error } = useUser()
-
-  useEffect(() => {
-    if (
-      !loading &&
-      !error &&
-      location.pathname !== "/orcid-link" &&
-      user?.provider === "google"
-    ) {
-      navigate("/orcid-link")
-    }
-  }, [location.pathname, user, loading, error, navigate])
-
-  if (loading || error) return null
-
-  const initialNotifications = user?.notifications?.map((n) => ({
-    id: n.id,
-    status: Array.isArray(n.notificationStatus)
-      ? n.notificationStatus.map((ns) => ns.status.toLowerCase())[0] || "unread"
-      : n.notificationStatus?.status?.toLowerCase() || "unread",
-    ...n,
-  })) || []
 
   return (
     <Uploader>
@@ -48,6 +22,7 @@ const Index: React.FC = () => {
               <HeaderContainer />
               <AppRoutes />
             </div>
+            <GoogleOrcidRedirect />
           </NotificationsProvider>
           <div className="sticky-footer">
             <FooterContainer />

--- a/packages/openneuro-app/src/scripts/index.tsx
+++ b/packages/openneuro-app/src/scripts/index.tsx
@@ -43,7 +43,7 @@ const Index: React.FC = () => {
     <Uploader>
       <SearchParamsProvider>
         <UserModalOpenProvider>
-          <NotificationsProvider initialNotifications={initialNotifications}>
+          <NotificationsProvider>
             <div className="sticky-content">
               <HeaderContainer />
               <AppRoutes />

--- a/packages/openneuro-app/src/scripts/users/notifications/user-notifications-context.tsx
+++ b/packages/openneuro-app/src/scripts/users/notifications/user-notifications-context.tsx
@@ -1,8 +1,18 @@
-import React, { createContext, useCallback, useContext, useState } from "react"
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from "react"
 import * as Sentry from "@sentry/react"
 import { useMutation } from "@apollo/client"
 import { UPDATE_NOTIFICATION_STATUS_MUTATION } from "../../queries/datasetEvents"
-import type { MappedNotification } from "../../types/event-types"
+import {
+  type MappedNotification,
+  mapRawEventToMappedNotification,
+} from "../../types/event-types"
+import { useUser } from "../../queries/user"
 
 interface NotificationsContextValue {
   notifications: MappedNotification[]
@@ -15,7 +25,7 @@ interface NotificationsContextValue {
 
 interface NotificationsProviderProps {
   children: React.ReactNode
-  initialNotifications?: MappedNotification[]
+  userId?: string
 }
 
 const NotificationsContext = createContext<
@@ -24,11 +34,16 @@ const NotificationsContext = createContext<
 
 export const NotificationsProvider: React.FC<NotificationsProviderProps> = ({
   children,
-  initialNotifications = [],
+  userId,
 }) => {
-  const [notifications, setNotifications] = useState<MappedNotification[]>(
-    initialNotifications,
-  )
+  const { user } = useUser(userId)
+  const [notifications, setNotifications] = useState<MappedNotification[]>([])
+
+  useEffect(() => {
+    const mapped = user?.notifications?.map(mapRawEventToMappedNotification) ??
+      []
+    setNotifications(mapped)
+  }, [user?.notifications])
   const [updateEventStatus] = useMutation(UPDATE_NOTIFICATION_STATUS_MUTATION)
 
   const handleUpdateNotification = useCallback(

--- a/packages/openneuro-app/src/scripts/users/notifications/user-notifications-view.tsx
+++ b/packages/openneuro-app/src/scripts/users/notifications/user-notifications-view.tsx
@@ -10,10 +10,6 @@ import styles from "./scss/usernotifications.module.scss"
 import iconUnread from "../../../assets/icon-unread.png"
 import iconSaved from "../../../assets/icon-saved.png"
 import iconArchived from "../../../assets/icon-archived.png"
-import { useUser } from "../../queries/user"
-import { Loading } from "../../components/loading/Loading"
-import * as Sentry from "@sentry/react"
-import { mapRawEventToMappedNotification } from "../../types/event-types"
 import {
   NotificationsProvider,
   useNotifications,
@@ -27,8 +23,6 @@ export const UserNotificationsView: React.FC<UserNotificationsViewProps> = (
   const { tab = "unread" } = useParams()
   const navigate = useNavigate()
   const location = useLocation()
-  const { user, loading, error } = useUser(orcidUser.id)
-
   const [indicatorStyle, setIndicatorStyle] = useState<React.CSSProperties>({
     width: "0px",
     transform: "translateX(0px)",
@@ -63,19 +57,8 @@ export const UserNotificationsView: React.FC<UserNotificationsViewProps> = (
     }
   }, [tab, orcidUser.orcid, navigate])
 
-  if (loading) return <Loading />
-  if (error) {
-    Sentry.captureException(error)
-    return (
-      <div>Error loading notifications: {error.message}. Please try again.</div>
-    )
-  }
-
-  const initialNotifications =
-    user?.notifications.map(mapRawEventToMappedNotification) || []
-
   return (
-    <NotificationsProvider initialNotifications={initialNotifications}>
+    <NotificationsProvider userId={orcidUser.id}>
       <div data-testid="user-notifications-view">
         <h3>Notifications for {orcidUser.name}</h3>
         <div className={styles.tabContainer}>

